### PR TITLE
Feature/cat 28 initial resource

### PR DESCRIPTION
**Pull request by brandonlee503**
Basically this week's deadline ticket CAT-28.

Here's the description:

-One get resource 
-Code in development complies 
-Backed by skeleton

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,53 +1,170 @@
...
+       “Courses taught”: “CS 160 … CS 161”
+}
+```
+
+###Dependencies
+Scraper:
+jsoup 1.8, tagsoup 1.2
+
+###Google Docs Version:
+https://docs.google.com/document/d/1y_Pyub3YOFrFQ0CYiEhIQdvlrPXRHliDVb-jiH84xM0/edit?usp=sharing
```
Comment (ghost) [inline readme.md]: is this google doc still accurate? if not, remove it from the readme.

File: readme.md [hunk 2]
```diff
@@ -1,53 +1,170 @@
...
+{
+“instructor”: “Brandon Lee”,
+        “rateMyProfessorLink”: “http://www.ratemyprofessors.com/”,
+       “Courses taught”: “CS 160 … CS 161”
+}
+```
+
+###Dependencies
+Scraper:
+jsoup 1.8, tagsoup 1.2
```
Comment (ghost) [inline readme.md]: are these scraper dependencies still necessary? if not, remove them.

File: catalogapi.yaml [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+      $ref: "#/definitions/Location"
+
+  Error:
+    type: string
+
+  Errors:
+    title: ArrayOfErrors
+    type: array
+    items:
+      $ref: "#/definitions/Error"
```
Comment (ghost) [inline catalogapi.yaml]: git complains about files without a trailing newline

File: readme.md [hunk 4]
```diff
@@ -1,53 +1,170 @@
...
+**Please note that if scraper is run again after any submitted PUT requests, those changes will be overwritten by the scraped data.
 
-    $ gradle idea
+Courses will be accessible by search through the course’s CRN.  For example CRN #50220 may reference one of the first computer science courses, ie. CS 101. And CRN #11933 may reference another course that follows, ie. CS 160.
 
-Open with `File` -> `Open Project`.
+### Setup Instructions
 
-### Build
+Open terminal, navigate to your cloned local directory, and enter the following line
```
Comment (ghost) [inline readme.md]: note that it's necessary to
1. `gradle build`, and
2. add credentials to configuration file

before you can run the application.

File: readme.md [hunk 5]
```diff
@@ -1,53 +1,170 @@
...
-
-    $ curl --include --request GET localhost:8008/
-    HTTP/1.1 200 OK
-    Date: Thu, 18 Jun 2015 22:11:47 GMT
-    Content-Type: text/plain
-    Content-Length: 11
-    
-    hello world
+{
+“CRN”: 11933,
```
Comment (ghost) [inline readme.md]: beware of curly quotes.

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/CatalogAPITestApplication.groovy [hunk 6]
```diff
@@ -0,0 +1,37 @@
+package edu.oregonstate.mist.catalogapitest
+
+import com.fasterxml.jackson.databind.SerializationFeature
+
```
Comment (ghost) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/CatalogAPITestApplication.groovy]: remove spaces between imports.

this is a suggestion, but i prefer to order imports starting with project dependencies (e.g. `edu.oregonstate.*`), then dropwizard dependencies (`io.dropwizard.*`), and then miscellaneous others.

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy [hunk 7]
```diff
@@ -0,0 +1,99 @@
+package edu.oregonstate.mist.catalogapitest.core
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+
```
Comment (browtayl) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy]: remove import space

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/db/CourseDAO.groovy [hunk 8]
```diff
@@ -0,0 +1,36 @@
+package edu.oregonstate.mist.catalogapitest.db
+
+import edu.oregonstate.mist.catalogapitest.core.Course
+import edu.oregonstate.mist.catalogapitest.mapper.CourseMapper
+
```
Comment (browtayl) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/db/CourseDAO.groovy]: remove import space

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/mapper/CourseMapper.groovy [hunk 9]
```diff
@@ -0,0 +1,28 @@
+package edu.oregonstate.mist.catalogapitest.mapper
+
+import edu.oregonstate.mist.catalogapitest.core.Course
+
+import org.skife.jdbi.v2.StatementContext
+import org.skife.jdbi.v2.tweak.ResultSetMapper
+
```
Comment (browtayl) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/mapper/CourseMapper.groovy]: remove import space

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+@XmlRootElement
+class Course {
+
+    private Integer cid
+    private Integer crn
+    private String courseName
+    private String instructor
+    private String day
+    private String time
+    private String location
```
Comment (ghost) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy]: in groovy, getters and setters are created implicitly for properties without access modifiers (public, private, protected).

you should be able to remove `private` from the properties above and then delete the corresponding getters and setters below.

for comparison: https://github.com/osu-mist/dropwizardtest/blob/master/src/main/groovy/edu/oregonstate/mist/dropwizardtest/core/Job.groovy

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+    public Course() {
+        // Jackson Deserialization
+    }
+
+    public Course(Integer cid, Integer crn, String courseName, String instructor, String day, String time, String location) {
```
Comment (ghost) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy]: suggestion: groovy also provides implicit constructors, so see if you can simplify or remove this.

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+    @JsonProperty
+    void setDay(day) {
+        this.day = day
+    }
+
+    @JsonProperty
+    void setTime(time) {
+        this.time = time
+    }
+}
```
Comment (ghost) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy]: add trailing newline to files where they are missing

File: readme.md [hunk 13]
```diff
@@ -1,53 +1,183 @@
...
-Build the project:
+```
+gradle build
+```
 
-    $ gradle build
+This builds the project into a single deployable jar file.
 
-JARs [will be saved](https://github.com/johnrengelman/shadow#using-the-default-plugin-task) into the directory `build/libs/`.
+Next you'll want to run the jar file along with your credentials in your config file.  To do so, enter the followwing line:
```
Comment (ghost) [inline readme.md]: typo

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/db/CourseDAO.groovy [hunk 14]
```diff
@@ -0,0 +1,35 @@
+package edu.oregonstate.mist.catalogapitest.db
+
+import edu.oregonstate.mist.catalogapitest.core.Course
+import edu.oregonstate.mist.catalogapitest.mapper.CourseMapper
+import org.skife.jdbi.v2.sqlobject.SqlUpdate
+import org.skife.jdbi.v2.sqlobject.SqlQuery
+import org.skife.jdbi.v2.sqlobject.Bind
+import org.skife.jdbi.v2.sqlobject.customizers.RegisterMapper
+
+
```
Comment (ghost) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/db/CourseDAO.groovy]: remove extra newline

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/mapper/CourseMapper.groovy [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+import org.skife.jdbi.v2.tweak.ResultSetMapper
+import java.sql.ResultSet
+import java.sql.SQLException
+
+public class CourseMapper implements ResultSetMapper<Course> {
+    public Course map(int i, ResultSet rs, StatementContext sc) throws SQLException {
+        Course course = new Course()
+
+        course.with {
+
```
Comment (ghost) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/mapper/CourseMapper.groovy]: remove newline

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy [hunk 16]
```diff
@@ -0,0 +1,17 @@
+package edu.oregonstate.mist.catalogapitest.core
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import javax.xml.bind.annotation.XmlRootElement
+
+@XmlRootElement
+class Course {
+
```
Comment (ghost) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy]: remove newline

File: src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy [hunk 17]
```diff
@@ -0,0 +1,17 @@
+package edu.oregonstate.mist.catalogapitest.core
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import javax.xml.bind.annotation.XmlRootElement
+
+@XmlRootElement
```
Comment (ghost) [inline src/main/groovy/edu/oregonstate/mist/catalogapitest/core/Course.groovy]: is `@XmlRootElement` necessary? if so, explain why. if not, remove it.

**Comment by ghost** [comment]
i reopened the jira task to document example usage using netcat in the readme

**Comment by ghost** [comment]
we talked about documenting in the readme only resources that are implemented

**Comment by ghost** [comment]
i am following the instructions in the readme and found that `javaCall.sh` is not in your git repo

**Comment by ghost** [comment]
In the readme, note that you need to download the oracle jar file.

**Comment by ghost** [comment]
the resource location in the "get by name" example needs to be percent encoded. i have made the necessary change.

**Pull request merged**